sys-block/gpart/patches/gpart-0.3~git.patchset

```diff
@@ -0,0 +1,74 @@
+From 95e9513621a19e11cae839ef75d55c605a7c2d3a Mon Sep 17 00:00:00 2001
+From: localanu <localanu@gmail.com>
+Date: Sat, 6 Jan 2018 10:46:19 +0000
+Subject: Add support for haiku
+
+diff --git a/src/disku.c b/src/disku.c
+index d50cebf..867def5 100644
+--- a/src/disku.c
++++ b/src/disku.c
+@@ -33,6 +33,10 @@
+ #include <sys/disk.h>
+ #endif
+ 
++#if defined(__HAIKU__)
++#include <Drivers.h>
++#endif
++
+ #include <unistd.h>
+ 
+ static void geometry_from_num_sectors(struct disk_geom *g, uint64_t nsects)
+@@ -95,8 +99,20 @@ static void os_disk_geometry(disk_desc *d, struct disk_geom *g)
+ 	g.d_nsecs = o / u;
+ 	g.d_c = g.d_nsecs / g.d_h / g.d_s;
+ }
++#elif defined(__HAIKU__)
++static void os_disk_geometry(disk_desc *d, struct disk_geom *g)
++{
++device_geometry dg;
++	if (ioctl(d->d_fd, B_GET_GEOMETRY, & dg) == 0 ){
```

Comment:
I think I should be more clear. [`ioctl()`][0] would return -1 on error, and any other value indicate success. This means that you should do this instead:

```diff
-+	if (ioctl(d->d_fd, B_GET_GEOMETRY, & dg) == 0 ){
++	if (ioctl(d->d_fd, B_GET_GEOMETRY, & dg) != -1){
```
[0]: http://pubs.opengroup.org/onlinepubs/009695399/functions/ioctl.html#tag_03_282_04